2_adventures/001/jonkernpa/04_day/code_walker.md

```diff
@@ -0,0 +1,35 @@
+# Finding Some Interesting Bits
+
+So, one of my gems that I enjoy -- because it just works -- is Dragonfly.
+
+The code in mongo_data_store has some really cool bits that are good to note:
+
+## Elmer Fudd's Memorization
+
+The memoized pattern (a lot like a singleton)
+
+    @var ||= create_instance_of_var
+
+Get the info the frst time. After that, return the same thing without the
+need to go get it again. A few examples:
+
+    @db ||= connection.db(database)
+    @grid ||= Mongo::Grid.new(db)
+    @authenticated ||= db.authenticate(username, password)
+
+## Hating on long constructor arg lists?
+
+Pass a hash!
+
+        def initialize(opts={})
```

Comment:
Yes.  Inline comments like this are meant to comment on the line above them, as that is what appears in pull requests etc.  Some people comment on the top, which is strange to me.  So my comment was simply about line 24.